src/Albany_ModelEvaluator.cpp

```diff
@@ -583,8 +583,6 @@ Thyra_OutArgs ModelEvaluator::createOutArgsImpl() const
         i + num_param_vecs,
         Thyra_ModelEvaluator::DERIV_LINEAR_OP);
 
-  result.setHessianSupports(true);
```

Comment:
What was happening until now for the Hessian of f? Was it computed by ROL with finite differences?

Reply:
This function call sets the AD support to true for all the Hessian computations including the Hessian-vector products of the response functions and the Hessian-vector products of the residual.

Previously, as the AD implementation of the Hessian computations is not finished for all cases (as you know, the Hessian-vector products of the residual are not supported yet for instance), I removed this function call and set the support individually.

To answer your question, for the Hessian-vector products of g, they were computed with AD by default.

However, this might have been the case for the Hessian-vector products of f too.